src/main/java/gift/service/TokenService.java

```diff
@@ -0,0 +1,62 @@
+package gift.service;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import java.security.Key;
+import java.util.Base64;
+import java.util.Date;
+import javax.crypto.spec.SecretKeySpec;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TokenService {
+
+    private final Key secretKey;
+
+    public TokenService(@Value("${jwt.secret}") String secretKey) {
+        byte[] decodedKey = Base64.getDecoder().decode(secretKey);
+        this.secretKey = new SecretKeySpec(decodedKey, 0, decodedKey.length, "HmacSHA256");
+    }
+
+    public String generateToken(String email) {
+        return Jwts.builder()
+            .setSubject(email)
+            .setIssuedAt(new Date())
+            .setExpiration(
+                new Date(System.currentTimeMillis() + 1000 * 60 * 60 * 10)) // 10 hours expiration
```

Comment:
주석 대신 코드로 나타내보는 것도 좋을 것 같아요.
```suggestion
                new Date(System.currentTimeMillis() + TEN_HOURS))
```

Reply:
static final long타입의 TEN_HOURS라는 이름을 가진 상수를 만들어 표현했습니다!